### PR TITLE
Fix softmax_backward_data cpu implementation error when argument output is noncontinguous

### PR DESCRIPTION
**Pull request by lixin-sxty**
Implementation of the `softmax_backward_data` operator for the CPU backend produces incorrect results when the `output` argument is non-contiguous. 

Here is a test case that demonstrates this issue:

```python
torch.manual_seed(0)
op = torch.ops.aten._softmax_backward_data
grad_output = torch.ones(3, 3, 3)
temp = torch.randn(3, 10, 3)
out = temp[:, :3, :]
out = out.contiguous()
print(out.is_contiguous())
grad_input = op(grad_output, out, 1, torch.float32)
print(grad_input)
```

In this test case, the variable `grad_input` yields incorrect results if the line `out = out.contiguous()` is commented out. With this fix, `grad_input` consistently produces the same results whenever `output` is contiguous.

### CODE REVIEW

**Comment by lixin-sxty** [comment]
Cc @scw @svenstaro @JackDanger This pull request encounters a check label failure, and the error message is unclear. Can you assist me? Thank you.

**Comment by svenstaro** [comment]
@lixin-sxty why ping me? I have nothing to do with this project.

**Comment by bdhirsh** [comment]
@lixin-sxty can you add a test, maybe in `test/test_torch.py`? Also - I see some noncontiguous tests for softmax_backward_data are skipped here, can you try removing that `skip()` and see if your fix causes those tests to pass? https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_methods_invocations.py#L14538

**Comment by lixin-sxty** [comment]
> @lixin-sxty can you add a test, maybe in `test/test_torch.py`? Also - I see some noncontiguous tests for softmax_backward_data are skipped here, can you try removing that `skip()` and see if your fix causes those tests to pass? https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_methods_invocations.py#L14538

I will check the tests for softmax_backward_data in common_methods_invocations.py and add a test case if necessary.

**Comment by lixin-sxty** [comment]
According to https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_methods_invocations.py#L14538, the test_noncontiguous_samples was failing on CPU. This case now passes with the fix, so I have removed this flag. @bdhirsh

**Comment by lixin-sxty** [comment]
@bdhirsh @mruberry Could you please review this change? Thank you!

**Review by zou3519 — APPROVED** [review]
